spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -692,6 +692,63 @@ public void multiRedisLockRegistryTest() throws InterruptedException, ExecutionE
 	}
 
 
+	@Test
+	@RedisAvailable
+	public void earlyWakeUpTest() throws InterruptedException {
+		final int THREAD_CNT = 3;
+		final String testKey = "testKey";
+
+		final CountDownLatch tryLockReady = new CountDownLatch(2);
+		final CountDownLatch awaitTimeout = new CountDownLatch(2);
+		final RedisConnectionFactory connectionFactory = getConnectionFactoryForTest();
+		final RedisLockRegistry registry1 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final RedisLockRegistry registry2 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final RedisLockRegistry registry3 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final ExecutorService executorService = Executors.newFixedThreadPool(THREAD_CNT);
+
+		Lock lock1 = registry1.obtain(testKey);
+		Lock lock2 = registry2.obtain(testKey);
+		Lock lock3 = registry3.obtain(testKey);
+		AtomicInteger expectOne = new AtomicInteger();
+
+		lock1.lock();
+		executorService.submit(() -> {
+			try {
+				tryLockReady.countDown();
+				boolean b = lock2.tryLock(3, TimeUnit.SECONDS);
```

Comment:
Changed to this `boolean b = lock2.tryLock(10, TimeUnit.SECONDS);`.

It is really possible to fail in the test with a race condition if we don't block thread long enough. 
There is no big enough time window between 2 and 3 seconds...